### PR TITLE
rec: Document negation of netmasks in settings.rst

### PR DESCRIPTION
**Pull request by nneul**
Add an example for negating dont-query

### Short description
Add example in doc for how to negate a netmask group - implements alternative solution for #12217 

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [NA] compiled this code
- [NA] tested this code
- [X] included documentation (including possible behaviour changes)
- [NA] documented the code
- [NA] added or modified regression test(s)
- [NA] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
Thanks, though I'd prefer a more general description of how to specify netmasks with negation and refer to that in relevant places.

File: pdns/recursordist/docs/settings.rst [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+   Anything else means **on**.
 
-As an example:
+   For example:
 
- - ``serve-rfc1918`` on its own means: do serve those zones.
- - ``serve-rfc1918=off`` or ``serve-rfc1918=no`` means: do not serve those zones.
- - Anything else means: do serve those zones.
+   - ``serve-rfc1918`` on its own means: do serve those zones.
+   - ``serve-rfc1918 = off`` or ``serve-rfc1918 = no`` means: do not serve those zones.
```
Comment (Habbie) [inline pdns/recursordist/docs/settings.rst:13]: Do we even allow whitespace around `=`?

File: pdns/recursordist/docs/settings.rst [hunk 2]
```diff
@@ -362,7 +371,7 @@ Any servers' name suffix-matching the supplied names will never be throttled.
 ----------------------------
 .. versionadded:: 4.2.0
 
--  Comma separated list of netmasks
+-  Comma separated list of netmasks, negation not supported
```
Comment (Habbie) [inline pdns/recursordist/docs/settings.rst:374]: what makes `dont-throttle-netmasks` so special that negation is not supported?

**Review by Habbie — CHANGES_REQUESTED** [review]

File: pdns/recursordist/docs/settings.rst (hunk 1 above)
Comment (omoerbeek) [reply pdns/recursordist/docs/settings.rst:13]: yes, the `boost::trim`` in https://github.com/PowerDNS/pdns/blob/05e6ab1c8a59243260f0f1f427d77363a87dd704/pdns/arguments.cc#L402

File: pdns/recursordist/docs/settings.rst (hunk 2 above)
Comment (Habbie) [reply pdns/recursordist/docs/settings.rst:374]: Ah, right. Perhaps that can go?

**Review by Habbie — APPROVED** [review]

File: pdns/recursordist/docs/settings.rst (hunk 2 above)
Comment (omoerbeek) [reply pdns/recursordist/docs/settings.rst:374]: LIkely yes, but I wanted docs only atm.

**Pull request merged**